tests/commands/test_get_nonce.py

```diff
@@ -0,0 +1,41 @@
+"""Tests for get-nonce command."""
+import logging
+import os
+from unittest.mock import patch
+
+import pytest
+
+from nile.utils.get_nonce import get_nonce
+
+GATEWAYS = {"localhost": "http://127.0.0.1:5050/"}
+
+
+@pytest.mark.parametrize(
+    "contract_address, network",
+    [("0xffff", "localhost"), ("0xffff", "goerli"), ("0xffff", "mainnet")],
+)
+@patch("nile.core.node.subprocess.check_output")
+@patch("nile.common.get_gateway", return_value=GATEWAYS)
+def test_call_format(mock_gateway, mock_subprocess, contract_address, network):
+    get_nonce(contract_address, network)
+
+    command = ["starknet", "get_nonce", "--contract_address", contract_address]
+    if network == "localhost":
+        command.append("--feeder_gateway_url=http://127.0.0.1:5050/")
+    else:
+        assert os.getenv("STARKNET_NETWORK") == f"alpha-{network}"
+
+    mock_subprocess.assert_called_once_with(command)
+
+
+@patch("nile.core.node.subprocess.check_output", return_value="5")
+@patch("nile.common.get_gateway", return_value=GATEWAYS)
+def test_get_nonce(mock_gateway, mock_subprocess, caplog):
```

Comment:
depending on what's decided re: my above comment, this could be moved to the cli's suite

Reply:
You mean the logging assertion, not the full test, right? The result assertion still makes sense in that case, I think.